### PR TITLE
don't use `gql` in `hook.js`

### PR DESCRIPTION
**Pull request by phryneas**
This moves the usage of `gql` out of the injected `hook.js` into the DevTools panel.

That should significantly reduce the amount of injected code.

image
🎉

### CODE REVIEW

File: src/application/components/Explorer/postMessageHelpers.ts [hunk 1]
```diff
@@ -84,7 +83,7 @@ export type ExplorerRequest = MessageEvent<{
   name: "ExplorerRequest";
   operationId: string;
   operation: string;
-  variables: JSONValue;
```
Comment (phryneas) [inline src/application/components/Explorer/postMessageHelpers.ts:87]: This is definitely never an array or a primitive value ;)

File: src/application/components/Explorer/Explorer.tsx [hunk 2]
```diff
@@ -43,20 +44,21 @@ function executeOperation({
...
-      variables,
-      fetchPolicy,
+    panelWindow.send({
+      type: "explorerRequest",
+      payload: {
+        operation: gql(operation),
+        operationName,
+        variables,
+        fetchPolicy,
+      },
```
Comment (phryneas) [inline src/application/components/Explorer/Explorer.tsx:59]: Actually, no need to `stringify` this one.

Everything here already came in from a `postMessage`, so it's transportable. The only possible exception would be the `DocumentNode` itself, but that seems to work nicely with `structuredClone`.

File: src/application/components/Explorer/Explorer.tsx [hunk 3]
```diff
@@ -220,7 +222,7 @@ export const Explorer = ({
           const observer = executeOperation({
             operation: event.data.operation,
             operationName: event.data.operationName,
-            variables: event.data.variables,
+            variables: event.data.variables ?? undefined,
```
Comment (phryneas) [inline src/application/components/Explorer/Explorer.tsx:225]: replacing a potential `null` with `undefined`.

**Review by jerelmiller — APPROVED** [review]
Mind adding a changeset for this one? Otherwise this looks great! What a great size savings here. Simple change but super effective 🚀

**Pull request merged**